staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/utils/BatchProgressTracker.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019-2022 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.services.staging.utils;
+
+import org.apache.commons.fileupload.ProgressListener;
+import org.apache.commons.fileupload.servlet.ServletFileUpload;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class BatchProgressTracker
+{
+    private BatchProgressTracker()
+    {
+        throw new IllegalStateException();
+    }
+
+    public static ConcurrentMap<String, Tracker> getInProgressTrackerMap()
+    {
+        return inProgressTrackerMap;
+    }
+
+    private static final ConcurrentHashMap<String, Tracker> inProgressTrackerMap = new ConcurrentHashMap<>();
+
+    public static void updateTracker(ServletFileUpload fileUpload)
+    {
+        fileUpload.setProgressListener(createProgressListener());
+    }
+
+    private static ProgressListener createProgressListener()
+    {
+        Tracker tracker = new Tracker();
+        tracker.setUploadStartTime(Instant.now());
+        return new ProgressListener()
+        {
+            private long megaBytes = -1;
```

Comment:
A proper class would be clearer in this case (and would also make debugging easier because the class will have a proper name):

```suggestion
        fileUpload.setProgressListener(new ProgressListenerImpl());
    }

    private static class ProgressListenerImpl implements ProgressListener
    {
        private final Tracker tracker;
        private long megaBytes;

        public ProgressListenerImpl()
        {
            final Tracker newTracker = new Tracker();
            newTracker.setUploadStartTime(Instant.now());

            this.tracker = newTracker;
            this.megaBytes = -1;
        }

```

Reply:
Done

Reply:
Incidentally I moved this nested class to a top-level class and renamed it to `BatchProgressListener` in commit 590f331c0db34ae520d5a2d8a635c62a07f8bd09.